Validations and some other things

HaydarAk opened the pull request:
Some minor additions for the Configuration Model, mainly idsm:validations.

JohannesLipp commented on model/communication/Endpoint.ttl:
```diff
@@ -51,6 +51,14 @@ ids:AppEndpoint
     rdfs:comment "Description of endpoints a DataApp offers."@en ;
+    idsm:validation [
+        idsm:forProperty ids:appEndpointPort;
+        idsm:constraint idsm:NotNull ;
```
The port declaration could be optional, as some might use default ports. You choose

HaydarAk replied:
Good point! Havent thought about that.

JohannesLipp commented on model/communication/Proxy.ttl:
```diff
@@ -18,6 +18,10 @@ ids:Proxy a owl:Class;
         idsm:constraint idsm:NotNull;
                   ];
+    idsm:validation [
+        idsm:forProperty ids:proxyAuthentication;
```
Please verify, if we need to enforce auth for any proxy. This might be optional.

JohannesLipp replied:
For userAuthentication below, we allow `idsm:OneToMany`. Could this be an option here?

HaydarAk replied:
You're right. A proxy could be used without authentication!

jlangkau commented on codes/RouteDeployMethod.ttl:
```diff
@@ -0,0 +1,14 @@
+idsc:APACHE_CAMEL a ids:RouteDeployMethod;
+    rdfs:label "Apache Camel"@en ;
+    rdfs:comment "Indicates that an AppRoute is deployed / implemented using Apache Camel."@en ;
+    .
```
Yes. I do it (very often) this way, too:

### EOF

HaydarAk commented:
Thanks @jlangkau  and @JohannesLipp for your help.
Added your recommendations in the last commit.

Additionally, as @jlangkau highlighted (in another channel), we should stay technology-agonistic for the routing related things too. Therefore I removed the Apache Camel instance and changed the property to range string for these kind of information.
